### PR TITLE
Add initial subscriber system

### PR DESCRIPTION
**Pull request by adambennett**
Proof of concept subscriber-like system for BaseUnityPlugins to hook into API behavior more easily, adding custom cards without the need to write patches at all.

Example usage: https://github.com/adambennett/MT-NyoxidesCardArchive/blob/1d67442cf67b4f7e4cdc1fae3369a14d8f560b93/Nyoxides%20Card%20Archive/CardArchive.cs#L36

### CODE REVIEW

File: MonsterTrainModdingAPI/Models/TrainModule.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+namespace MonsterTrainModdingAPI.Models
+{
+    public interface TrainModule
+    {
+        List<CardDataBuilder> RegisterCustomCards();
+        List<RelicData> RegisterCustomArtifacts();
+        List<CharacterData> RegisterCustomCharacters();
+        BaseUnityPlugin RegisterPlugin();
+    }
+}
```
Comment (RawsomeGH) [inline MonsterTrainModdingAPI/Models/TrainModule.cs:14]: Missing newline at end of file

File: MonsterTrainModdingAPI/MonsterTrainModdingAPI.cs [hunk 2]
```diff
@@ -14,12 +18,33 @@ namespace MonsterTrainModdingAPI
...
+        private void Initialize()
+        {
+            FillPluginsList();
+            CustomCardManager.RegisterAllCustomCards();
+        }
+        
+        private void FillPluginsList()
+        {
+            plugins = new List<TrainModule>();
+            // Somebody figure out how to fill in this list of mods right here
```
Comment (RawsomeGH) [inline MonsterTrainModdingAPI/MonsterTrainModdingAPI.cs:47]: :monstertrainhaha:

File: MonsterTrainModdingAPI/builders/CardDataBuilder.cs [hunk 3]
```diff
@@ -44,6 +46,8 @@ public class CardDataBuilder
         public CardData.CostType CostType { get; set; }
         public FallbackData FallbackData { get; set; }
 
+        public MTClan Clan;
```
Comment (KittenAqua) [inline MonsterTrainModdingAPI/builders/CardDataBuilder.cs:49]: This should be a property, like all of the ones above it. Just add `{ get; set; }` after it.

File: MonsterTrainModdingAPI/Models/TrainModule.cs (hunk 1 above)
Comment (adambennett) [reply MonsterTrainModdingAPI/Models/TrainModule.cs]: file deleted

File: MonsterTrainModdingAPI/builders/CardDataBuilder.cs (hunk 3 above)
Comment (adambennett) [reply MonsterTrainModdingAPI/builders/CardDataBuilder.cs]: Fixed

**Pull request merged**